picard/ui/options/scripting.py

```diff
@@ -106,10 +123,141 @@ def __init__(self, parent=None):
+        a Picard script package.
+        """
+        FILE_ERROR_IMPORT = N_('Error importing "%s". %s.')
+        FILE_ERROR_DECODE = N_('Error decoding "%s". %s.')
+
+        dialog_title = _("Import Script File")
+        dialog_file_types = self._get_dialog_filetypes()
+        options = QtWidgets.QFileDialog.Options()
+        options |= QtWidgets.QFileDialog.DontUseNativeDialog
+        filename, file_type = QtWidgets.QFileDialog.getOpenFileName(self, dialog_title, self.default_script_directory, dialog_file_types, options=options)
+        if filename:
+            log.debug('Importing tagging script file: %s' % filename)
+            try:
+                with open(filename, 'r', encoding='utf8') as i_file:
+                    file_content = i_file.read()
+            except OSError as error:
+                self.output_file_error(FILE_ERROR_IMPORT, filename, error.strerror)
+                return
+            if not file_content.strip():
+                self.output_file_error(FILE_ERROR_IMPORT, filename, _('The file was empty'))
+                return
+            if file_type == self.FILE_TYPE_PACKAGE:
+                try:
+                    script_item = PicardScript().create_from_yaml(file_content)
+                except ScriptImportError as error:
+                    self.output_file_error(FILE_ERROR_DECODE, filename, error)
+                    return
+            else:
+                script_item = PicardScript(
+                    title=_("Imported from %s") % filename,
+                    script=file_content.strip()
+                )
+            title = _("%s (imported)") % script_item["title"]
+            list_item = ScriptListWidgetItem(title, False, script_item["script"])
+            self.ui.script_list.addItem(list_item)
+            self.ui.script_list.setCurrentRow(self.ui.script_list.count() - 1)
+
+    def export_script(self):
+        """Export the current script to an external file. Export can be either as a plain text
+        script or a naming script package.
+        """
+        FILE_ERROR_EXPORT = N_('Error exporting file "%s". %s.')
+
+        items = self.ui.script_list.selectedItems()
+        if not items:
+            return
+
+        item = items[0]
+        script_text = item.script
+        script_title = item.name if item.name.strip() else _("Unnamed Script")
+
+        if script_text:
+            script_item = PicardScript(title=script_title, script=script_text)
+            script_filename = ".".join((script_item.filename, self.default_script_extension))
+            default_path = os.path.normpath(os.path.join(self.default_script_directory, script_filename))
+            dialog_title = _("Export Script File")
+            dialog_file_types = self._get_dialog_filetypes()
+            options = QtWidgets.QFileDialog.Options()
+            options |= QtWidgets.QFileDialog.DontUseNativeDialog
```

Comment:
Is there any specific reason this does not allow the native dialog? Usually we should prefer the native one, unless we need some functionality that only the Qt dialog provides.

Reply:
No reason.  Just got caught up with a copy'n'paste from elsewhere in the code.  I'll fix that.